bot/university/university.py

```diff
@@ -128,6 +143,293 @@ def check_reaction(_reaction, user):
+
+        Deletes the message in the exchange channel as well as all corresponding entries in the db.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            channel (discord.TextChannel): The channel corresponding to the course.
+        """
+        message_id = self._db_connector.get_group_exchange_message(ctx.author.id, channel.id)
+        if message_id is not None:
+            self._db_connector.remove_group_exchange_offer(ctx.author.id, channel.id)
+            msg = await ctx.guild.get_channel(constants.CHANNEL_ID_GROUP_EXCHANGE).fetch_message(message_id)
+            await msg.delete()
+
+    async def _notfiy_author_about_candidates(self,
+                                              author: discord.User,
+                                              potential_candidates: List[Tuple[str, str, int]],
+                                              channel: discord.TextChannel,
+                                              course_channel: discord.TextChannel):
+        """Notifies the Author of a group exchange request about possible exchange candidates.
+
+        The author is informed via a direct message which contains infos about all possible users he or she could
+        exchange groups with.
+
+        Args:
+            author (discord.User): The author to be notified.
+            potential_candidates (List[Tuple[str, str]]): The possible candidate ids and the message ids of their
+            exchange messages.
+            channel (discord.TextChannel): The channel in which the the message of potential candidates can be found
+            course_channel (discord.TextChannel): The channel that refers to the course that the exchange is for.
+        """
+        embed = discord.Embed(title="Wir haben potentielle Tauschpartner für dich gefunden:",
+                              description="Bitte vergiss nicht, deine Anfrage mit dem Befehl 'exchange remove "
+                                          "<channel-mention>' wieder zu löschen, wenn du einen Tauschpartner gefunden "
+                                          "hast",
+                              color=constants.EMBED_COLOR_GROUP_EXCHANGE)
+        user_string = ""
+        group_string = ""
+        course_name = "- " + _parse_course_from_channel_name(course_channel)
+        course_name = "\n".join(course_name for _ in potential_candidates)
+
+        for candidate in potential_candidates:
+            user = self.bot.get_user(int(candidate[0]))
+            msg = await channel.fetch_message(int(candidate[1]))
+            group = candidate[2]
+            user_string += "- [{0}]({1})\n".format(user, msg.jump_url)
+            group_string += "- Gruppe {0}\n".format(group)
+
+        embed.add_field(name="User:", value=user_string) \
+            .add_field(name="Kurs:", value=course_name) \
+            .add_field(name="Bietet:", value=group_string)
```

Comment:
Yeah that's true, I didn't think of that. 
I still dont like that the distances are inconsistent between the users, even if I used just one group. 
But then it would be 

Pklempe: 2
Kompe: 4
Brandx103: 2
...

Reply:
The more important information here is the group number. That's why I would like to have it first. It also makes it easier to read because the important part is always the same regarding look and position. The difference in lengths of the nicknames is the reason for me to keep them at the end of the line.